plugins/c8y_configuration_plugin/src/main.rs

```diff
@@ -73,124 +63,99 @@ pub struct ConfigPluginOpt {
-        return Ok(());
-    }
-
     // Load tedge config from the provided location
     let tedge_config_location =
         tedge_config::TEdgeConfigLocation::from_custom_root(&config_plugin_opt.config_dir);
-    let config_repository = tedge_config::TEdgeConfigRepository::new(tedge_config_location.clone());
-
     let log_level = if config_plugin_opt.debug {
         tracing::Level::TRACE
     } else {
         get_log_level(
-            "c8y-configuration-plugin",
+            "c8y-log-plugin",
             tedge_config_location.tedge_config_root_path.to_path_buf(),
         )?
     };
+
     set_log_level(log_level);
 
+    let config_repository = tedge_config::TEdgeConfigRepository::new(tedge_config_location.clone());
     let tedge_config = config_repository.load()?;
 
-    let tedge_device_id = tedge_config.query(DeviceIdSetting)?;
+    if config_plugin_opt.init {
+        init(config_plugin_opt.config_dir)
+    } else {
+        run(tedge_config).await
+    }
+}
 
-    let mqtt_host = tedge_config.query(MqttClientHostSetting)?;
-    let mqtt_port = tedge_config.query(MqttClientPortSetting)?.into();
-    let http_client = create_http_client(&tedge_config).await?;
-    let http_client: Arc<Mutex<dyn C8YHttpProxy>> = Arc::new(Mutex::new(http_client));
-    let tmp_dir = tedge_config.query(TmpPathSetting)?.into();
-    let data_dir: PathBuf = tedge_config.query(DataPathSetting)?.into();
-
-    let http_port: u16 = tedge_config.query(HttpPortSetting)?.into();
-    let http_address = tedge_config.query(HttpBindAddressSetting)?.to_string();
-    let local_http_host = format!("{}:{}", http_address, http_port);
-
-    let mut config_manager = ConfigManager::new(
-        tedge_device_id,
-        mqtt_host,
-        mqtt_port,
-        http_client,
-        local_http_host,
-        tmp_dir,
-        config_plugin_opt.config_dir,
-        data_dir.join(DEFAULT_FILE_TRANSFER_DIR_NAME),
-    )
-    .await?;
-
-    config_manager.run().await
+async fn run(tedge_config: TEdgeConfig) -> Result<(), anyhow::Error> {
+    let runtime_events_logger = None;
+    let mut runtime = Runtime::try_new(runtime_events_logger).await?;
+
+    // Create actor instances
+    let mqtt_config = mqtt_config(&tedge_config)?;
+    let mut jwt_actor = C8YJwtRetriever::builder(mqtt_config.clone());
+    let mut http_actor = HttpActor::new().builder();
+    let c8y_http_config = (&tedge_config).try_into()?;
+    let mut c8y_http_proxy_actor =
+        C8YHttpProxyBuilder::new(c8y_http_config, &mut http_actor, &mut jwt_actor);
+
+    let mut fs_watch_actor = FsWatchActorBuilder::new();
+    let mut signal_actor = SignalActor::builder();
+    let mut timer_actor = TimerActor::builder();
+
+    //Instantiate health monitor actor
+    let mut health_actor = HealthMonitorBuilder::new(PLUGIN_NAME);
+    let mqtt_config = health_actor.set_init_and_last_will(mqtt_config);
+    let mut mqtt_actor = MqttActorBuilder::new(mqtt_config.clone().with_session_name(PLUGIN_NAME));
```

Comment:
As this has to done for any main function using the health actor, one can consider in a follow-up task to provide an `HealthMonitorBuilder` method that connect to the MQTT actor updating the MQTT config.